node/service/src/relay_chain_selection.rs

```diff
@@ -417,8 +417,8 @@ where
 
 		let (lag, subchain_head) = if cfg!(feature = "disputes") {
 			// Prevent sending flawed data to the dispute-coordinator.
-			if Some(subchain_block_descriptions.len() as _) !=
-				subchain_number.checked_sub(target_number)
+			if Some(subchain_block_descriptions.len() as u32) !=
+				(subchain_number + 1).checked_sub(target_number)
```

Comment:
That looks wrong. I think this should be `subchain_number.checked_sub(target_number + 1)`. As it should be the ancestors of the block with `subchain_number`. Given subchain_number 10 and target_number 9, there would be no ancestor, so the result should be 0.

Reply:
Not 100% as the code is quite involved.

Reply:
I took @Lldenaurois ' initial and added `+1` on each side of the inequality sign Wich is a sane transform. But we should add the rationale for the initial `-1` for sure to argue on solid grounds :)

Reply:
Will re-check tomorrow.